batch7crud-roll9/src/main/java/com/lftechnology/batch7crud/services/StudentService.java

```diff
@@ -12,8 +15,13 @@
 public class StudentService {
   private StudentDAO stdDAO = new StudentDAO();
 
-  public Student save(Student student) throws DataException {
-    return stdDAO.insert(student);
+  public HashMap save(Student student) throws DataException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The return type of this method should be an interface such as "Map" rather than the implementation "HashMap". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1319)